scipy/stats/_stats_py.py

```diff
@@ -4363,6 +4366,10 @@ def confidence_interval(self, confidence_level=0.95, method=None):
         return ci
 
 
+xp_capabilities_table['pearsonr'] = dict(cpu_only=True)
```

Comment:
Currently, `xp_capabilities_table` is a dictionary defined in the module per Ralf's suggestion, but it can be defined anywhere.

The keys are function names, and the values are dictionaries that accept keyword input similar to the `skip_xp_backend` decorator, like `cpu_only`, `np_only`, and `exceptions`. The one difference is that all the backends to skip and corresponding reasons would be provided as list of tuples, e.g.
```python3
[('cupy', 'I said so'), 
 ('jax.numpy', 'Just because']
```
Or I suppose these tuple could be provided as `*args` (each tuple is an element), but currently it's a single argument, `skip_backends`.

I've only implemented `skip_backends` and `cpu_only` for now.

Reply:
Looks like we'd need mark `jax.numpy` support as false since all tests are skipped due to item assignment so a good opportunity to test this out